src/execution/join_hashtable.cpp

```diff
@@ -275,32 +270,156 @@ void JoinHashTable::Finalize() {
+			conflict_entries.push_back(pointers[index]);
+		}
+		// replace the hash_value in the current entry and point to a position in the hash_map
+		Store<data_ptr_t>(pointers[index], next_ptr);
+		// store the pointer to the current tuple entry in the hash_map
+		pointers[index] = key_locations[i];
+	}
+	// It is still necessary to handle multiple conflicts to the same key
+	if (has_primary_key) {
+		for (auto entry : conflict_entries) {
+			auto next_entry_ptr = Load<data_ptr_t>(entry + pointer_offset);
+			// check the whole chain
+			while (has_primary_key && next_entry_ptr != nullptr) {
+				// check whether the keys are the same
+				for (idx_t key_idx = 0; key_idx != condition_types.size(); ++key_idx) {
+					auto key_type = condition_types[key_idx];
+					auto key_offset = col_offsets[key_idx];
+					has_primary_key = !CompareKeysSwitch(entry + key_offset, next_entry_ptr + key_offset, key_type);
+				}
+				// walk to the next entry
+				next_entry_ptr = Load<data_ptr_t>(next_entry_ptr + pointer_offset);
+			}
+			// no more conflicts for this key
+		}
+	}
+}
+
+template <typename T>
+bool TemplatedKeysCompare(data_ptr_t left_ptr, data_ptr_t right_ptr) {
+	const auto left_val = Load<T>(left_ptr);
+	const auto right_val = Load<T>(right_ptr);
+	return Equals::Operation<T>(left_val, right_val);
+}
+
+bool JoinHashTable::CompareKeysSwitch(data_ptr_t left_key, data_ptr_t right_key, const LogicalType &key_type) {
+
+	if (key_type.id() == LogicalTypeId::LIST) {
+		return true; // not handling this type for now, return as if they were equals
+	}
+	switch (key_type.InternalType()) {
+	case PhysicalType::BOOL:
+	case PhysicalType::INT8:
+		return TemplatedKeysCompare<int8_t>(left_key, right_key);
+	case PhysicalType::INT16:
+		return TemplatedKeysCompare<int16_t>(left_key, right_key);
+	case PhysicalType::INT32:
+		return TemplatedKeysCompare<int32_t>(left_key, right_key);
+	case PhysicalType::INT64:
+		return TemplatedKeysCompare<int64_t>(left_key, right_key);
+	case PhysicalType::UINT8:
+		return TemplatedKeysCompare<uint8_t>(left_key, right_key);
+	case PhysicalType::UINT16:
+		return TemplatedKeysCompare<uint16_t>(left_key, right_key);
+	case PhysicalType::UINT32:
+		return TemplatedKeysCompare<uint32_t>(left_key, right_key);
+	case PhysicalType::UINT64:
+		return TemplatedKeysCompare<uint64_t>(left_key, right_key);
+	case PhysicalType::INT128:
+		return TemplatedKeysCompare<hugeint_t>(left_key, right_key);
+	case PhysicalType::FLOAT:
+		return TemplatedKeysCompare<float>(left_key, right_key);
+	case PhysicalType::DOUBLE:
+		return TemplatedKeysCompare<double>(left_key, right_key);
+	case PhysicalType::INTERVAL:
+		return TemplatedKeysCompare<interval_t>(left_key, right_key);
+	case PhysicalType::VARCHAR:
+		return TemplatedKeysCompare<string_t>(left_key, right_key);
+	case PhysicalType::LIST:
+	case PhysicalType::MAP:
+	case PhysicalType::STRUCT:
+		return true; // not handling this type for now, return as if they were equals
```

Comment:
Should it not default to return `false`, and just skip the primary key optimization for now? Or rather detect this earlier on so this function is never called, and throw an exception if this type is used at all?